src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java

```diff
@@ -342,6 +351,21 @@ protected List<Map<String, String>> csvRowFieldValueMapping(
         .toList();
   }
 
+  public Set<Map<String, String>> checkForDuplicates(
+      Collection<Map<String, String>> rows, String entityDescriptor) {
+    Set<Map<String, String>> rowsSet = new HashSet<>(rows);
```

Comment:
Maybe in `EntityData` we need a method to tell us unique fields to check?
So for `UniqueEntity`, this would be `[['uuid'],['id']]` because both uuid and id need to be unique on their own.
And for some timeseries entry of some kind this would maybe be `[['inputModel', 'time']]` or just `[['time']]`, if it's just a time series for a single source

Reply:
I think I found a good solution for this.